common/util.go

```diff
@@ -263,6 +263,14 @@ func CheckDecisionResultLimit(
 	return nil
 }
 
+// ToServiceTransientError converts an error to ServiceTransientError
+func ToServiceTransientError(err error) error {
```

Comment:
Do you mind adding a unit test roundtripping this through `IsServiceTransientError`?